File: android/app/src/main/java/com/tisza/esemenynaptar/database/EventDao.java
package com.tisza.esemenynaptar.database;

import androidx.lifecycle.*;
import androidx.room.*;
import androidx.sqlite.db.*;

import java.util.*;

@Dao
public interface EventDao
{
	@Query("SELECT * FROM event WHERE year = :year AND month = :month AND day = :day")
	LiveData<List<Event>> getEventsForDate(int year, int month, int day);

	default LiveData<List<Event>> getEventsForDate(Calendar calendar)
	{
		return getEventsForDate(calendar.get(Calendar.YEAR), calendar.get(Calendar.MONTH), calendar.get(Calendar.DAY_OF_MONTH));
	}

	@Update
	void updateEvent(Event event);
}

File: android/app/src/main/java/com/tisza/esemenynaptar/database/EventDao.kt
package com.tisza.esemenynaptar.database

import androidx.lifecycle.*
import androidx.room.*
import java.util.*

@Dao
interface EventDao {
    @Query("SELECT * FROM event WHERE year = :year AND month = :month AND day = :day")
    fun getEventsForDate(year: Int, month: Int, day: Int): LiveData<List<Event>>

    @Query("SELECT * FROM event WHERE isLiked ORDER BY year, month, day")
    fun getLikedEvents(): LiveData<List<Event>>

    @Update
    fun updateEvent(event: Event?)
}

fun EventDao.getEventsForDate(calendar: Calendar): LiveData<List<Event>> {
    return getEventsForDate(calendar[Calendar.YEAR], calendar[Calendar.MONTH], calendar[Calendar.DAY_OF_MONTH])
}
